ddocs/medic-db/medic-sms/views/gateway_messages_by_state/map.js

```diff
@@ -0,0 +1,22 @@
+function (doc) {
+  var _emit = function(tasks) {
+    tasks.forEach(function(task) {
+      if (!task.messages || !task.gateway_ref) {
+        return;
+      }
+
+      task.messages.forEach(function(msg) {
+        if (msg.uuid && msg.to && msg.message) {
+          var value = {
+            id: msg.uuid,
+            gateway_ref: task.gateway_ref,
+          };
+          emit(task.state, value);
+
+        }
+      });
+    });
+  };
+  _emit(doc.tasks || []);
+  _emit(doc.scheduled_tasks || []);
+}
```

Comment:
It's a shame the `medic/messages_by_state` view can't be retrofitted to do this too...

Reply:
Yea. `messages_by_state` is used for exporting (with skip and limit and no other filters), so adding an additional emit per message would effectively duplicate every message in the export :( Oh, I think this already happens because of the 'pending-or-forwarded' additional emit, but only for messages in that state. Oh well. 

We also use `messages_by_state` with start_key + end_key, so emitting a similar additional key would produce duplicates there too, so I'd have to keep a simple, a non-array key, like this new view. 

It would have just made  `messages_by_state` too bloated IMO.